components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/IdentityProviderDataRetrievalClient.java

```diff
@@ -102,6 +112,131 @@ public String getIdPImage(String tenant, String idpName)
         return StringUtils.EMPTY;
     }
 
+    /**
+     * This function retrieves the given configurations of the given federated identity provider using carbon API.
+     * Three HTTP GET calls will be executed as below.
+     * 1. <host>/t/carbon.super/api/server/v1/identity-providers?filter=name+eq+Google
+     * 2. <host>/t/carbon.super/api/server/v1/identity-providers/6719c5cc-5162-44c7-9190-cb74d500f5fc
+     * 3. <host>/t/carbon.super/api/server/v1/identity-providers/6719c5cc-5162-44c7-9190-cb74d500f5fc/
+     * federated-authenticators/R29vZ2xlT0lEQ0F1dGhlbnRpY2F0b3I
+     *
+     * @param tenant     String. The tenant name of the session i.e. carbon.super.
+     * @param idpCode    String. The declared identity provide code i.e. GoogleOIDCAuthenticator.
+     * @param idpName    String. The identity provider name setup at carbon console i.e. Google.
+     * @param configKeys List<String>. The list of keys of required configurations i.e. ClientId, callbackUrl
```

Comment:
Let's end the comments with a `.`. Let's fix other places as well.

Reply:
Fixed 38d0c51fd441a7ee513b0b52d2459c3a732d9e2d